src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -971,6 +977,94 @@ private void ReadCompressedTextChunk(PngMetadata metadata, ReadOnlySpan<byte> da
             {
                 metadata.TextData.Add(new PngTextData(name, uncompressed, string.Empty, string.Empty));
             }
+
+            if (name.Equals("Raw profile type exif", StringComparison.OrdinalIgnoreCase))
+            {
+                this.ReadLegacyExifTextChunk(baseMetadata, uncompressed);
+            }
+        }
+
+        /// <summary>
+        /// Reads exif data encoded into a text chunk with the name "raw profile type exif".
+        /// This method was used by ImageMagick, exiftool, exiv2, digiKam, etc, before the
+        /// 2017 update to png that allowed a true exif chunk.
+        /// </summary>
+        /// <param name="metadata">The <see cref="ImageMetadata"/> to store the decoded exif tags into.</param>
+        /// <param name="data">The contents of the "raw profile type exif" text chunk.</param>
+        private void ReadLegacyExifTextChunk(ImageMetadata metadata, string data)
+        {
+            ReadOnlySpan<char> dataSpan = data.AsSpan();
+            dataSpan = dataSpan.TrimStart();
+
+            if (!dataSpan.Slice(0, 4).ToString().Equals("exif", StringComparison.OrdinalIgnoreCase))
+            {
+                // "exif" identifier is missing from the beginning of the text chunk
+                return;
+            }
+
+            // Skip to the data length
+            dataSpan = dataSpan.Slice(4).TrimStart();
+            int dataLengthEnd = dataSpan.IndexOf('\n');
+            int dataLength = int.Parse(dataSpan.Slice(0, dataSpan.IndexOf('\n')).ToString());
+
+            // Skip to the hex-encoded data
+            dataSpan = dataSpan.Slice(dataLengthEnd).Trim();
+            string dataSpanString = dataSpan.ToString().Replace("\n", string.Empty);
+            if (dataSpanString.Length != (dataLength * 2))
+            {
+                // Invalid length
+                return;
+            }
+
+            // Parse the hex-encoded data into the byte array we are going to hand off to ExifProfile
+            byte[] dataBlob = new byte[dataLength - ExifHeader.Length];
```

Comment:
Is this array allocation OK?
(just perf-wise)

Reply:
```suggestion
            using IMemoryOwner<byte> dataBuffer = this.memoryAllocator.Allocate<byte>(dataLength);
            Span<byte> dataBlob = dataBuffer.GetSpan();
```

After #1730, I'm planning to create a utility similar to SixLabors/Font's [Buffer<T>](https://github.com/SixLabors/Fonts/blob/master/src/SixLabors.Fonts/Buffer%7BT%7D.cs) that avoids `IMemoryOwner` allocation when it's backed by `ArrayPool<byte>.Shared`, but for now this should be good enough.

Reply:
@antonfirsov this suggestion would be ideal, however currently `ExifProfile` does not have a constructor that can accept `Memory`/`Span` objects, only `byte[]`.

Reply:
Ah I missed that it goes to `ExifProfile`, than it should be a `new byte[]` indeed.